dnm: debug rnot

bharathv opened the pull request:
<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes
* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

bharathv committed:
debug rnot

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=20
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75122-->
<!--e2924bad-9fc9-47a2-bdb3-c239cba20151-->
<!--title-->
## Retry command for [Build#75122](https://buildkite.com/redpanda/redpanda/builds/75122)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest.test_node_operations@{"failures":true}
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest.test_node_operations@{"failures":true}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--b8ed4ece-3bb4-456f-af5a-1c70fd3fed38-->
<!--TEST RESULTS-->
<!--f6f7e07f-eb49-4c18-b58b-1e09cb456735-->
<!--TEST RESULTS-->
<!--1011c17c-9383-40e4-9f20-89f2dcb283d8-->
<!--TEST RESULTS-->
<!--cf0df6b7-3bc8-4f26-8b15-533c7b32d6f1-->
<!--TEST RESULTS-->
<!--a5bcac84-a048-4705-b8f7-d04555eb7c71-->
<!--TEST RESULTS-->
<!--7c900fff-864d-4c94-b6ff-ca7c3c98d0a0-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75122">build#75122</a></summary>


| test_class                 | test_method          | test_arguments     | test_kind   | job_url                                                                                   | test_status   | passed   | reason   | test_history                                                                                                                                                  |
|:---------------------------|:---------------------|:-------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75122#019a2bac-8aa0-40c6-ac9b-fec5ddc6d7bf | FLAKY         | 18/20    |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75160">build#75160</a></summary>


| test_class                 | test_method          | test_arguments     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                     | test_history                                                                                                                                                  |
|:---------------------------|:---------------------|:-------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75160#019a2c97-ca19-412b-baaa-630c575b148e | FLAKY         | 57/60    | upstream reliability is '98.4'. current run reliability is '86.36363636363636'. drift is 12.03636 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75160#019a2c97-ca1d-4e83-bd1f-f0b1b42b5d6a | FLAKY         | 57/60    | upstream reliability is '98.4'. current run reliability is '86.36363636363636'. drift is 12.03636 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75174">build#75174</a></summary>


| test_class                 | test_method          | test_arguments     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                  |
|:---------------------------|:---------------------|:-------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75174#019a2dd9-ae1d-4046-9063-bbacc0e40550 | FLAKY         | 59/60    | upstream reliability is '98.14814814814815'. current run reliability is '95.23809523809523'. drift is 2.91005 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75174#019a2dd9-ae25-4964-be4c-f1b838ce1c99 | FLAKY         | 58/60    | upstream reliability is '98.40848806366049'. current run reliability is '90.47619047619048'. drift is 7.9323 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75185">build#75185</a></summary>


| test_class                                                     | test_method                                                    | test_arguments     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                 | test_history                                                                                                                                                  |
|:---------------------------------------------------------------|:---------------------------------------------------------------|:-------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingRandomOpsTest                                     | test_node_operations                                           | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75185#019a2ee2-67f9-4e90-9e40-7c8fff2fe2dd | FAIL          | 0/60     | The test has failed across all retries | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| src/v/cluster_link/replication/tests/link_replication_mgr_test | src/v/cluster_link/replication/tests/link_replication_mgr_test |                    | unit        | https://buildkite.com/redpanda/redpanda/builds/75185#019a2ec4-6c6c-47bc-b235-2b810254c91f | FAIL          | 0/1      |                                        |                                                                                                                                                               |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75250">build#75250</a></summary>


| test_class                 | test_method          | test_arguments     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                  |
|:---------------------------|:---------------------|:-------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75250#019a31a5-ed5b-4877-9c64-38ba111bfdf8 | FLAKY         | 58/60    | upstream reliability is '98.41986455981942'. current run reliability is '90.47619047619048'. drift is 7.94367 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75250#019a31bc-b254-4d41-9956-03cb01ee4ae3 | FLAKY         | 58/60    | upstream reliability is '98.41986455981942'. current run reliability is '90.9090909090909'. drift is 7.51077 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75268">build#75268</a></summary>


| test_class                 | test_method          | test_arguments     | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                   | test_history                                                                                                                                                  |
|:---------------------------|:---------------------|:-------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feb0-4e97-9a1a-4897b115b3d7 | FLAKY         | 59/60    | upstream reliability is '69.87179487179486'. current run reliability is '95.23809523809523'. drift is -25.3663 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feb1-4656-b17c-cb73cb911ef3 | FLAKY         | 57/60    | upstream reliability is '69.87179487179486'. current run reliability is '85.71428571428571'. drift is -15.84249 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feb1-4967-807b-8b439ec08c5f | FLAKY         | 58/60    | upstream reliability is '69.87179487179486'. current run reliability is '90.9090909090909'. drift is -21.0373 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feb2-41dc-80d8-cab1068e9124 | FLAKY         | 57/60    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feb3-4e70-9603-a62066e32976 | FLAKY         | 58/60    | upstream reliability is '98.41986455981942'. current run reliability is '90.9090909090909'. drift is 7.51077 and the allowed drift is set to 50. The test should PASS    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feb9-44d2-9a90-62ca3a2a6b94 | FLAKY         | 59/60    | upstream reliability is '98.41986455981942'. current run reliability is '95.23809523809523'. drift is 3.18177 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feba-45f4-ba36-c584855b9b4b | FLAKY         | 58/60    | upstream reliability is '98.41986455981942'. current run reliability is '90.9090909090909'. drift is 7.51077 and the allowed drift is set to 50. The test should PASS    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feba-4d3f-87fb-f7bac0710324 | FLAKY         | 57/60    | upstream reliability is '98.41986455981942'. current run reliability is '86.95652173913044'. drift is 11.46334 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-feba-4f38-b98b-60d2a504a2d8 | FLAKY         | 59/60    | upstream reliability is '69.87179487179486'. current run reliability is '95.23809523809523'. drift is -25.3663 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-febb-4a0b-b06c-67cfa4d0ae6f | FLAKY         | 39/40    |                                                                                                                                                                          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-febb-4dd3-b17e-2afb2099533b | FLAKY         | 57/60    | upstream reliability is '98.41986455981942'. current run reliability is '86.95652173913044'. drift is 11.46334 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-febb-4e2a-86ce-15b0b1c6ebde | FLAKY         | 57/60    | upstream reliability is '98.41986455981942'. current run reliability is '86.36363636363636'. drift is 12.05623 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-febc-40a9-ac65-6d58ec375985 | FLAKY         | 57/60    | upstream reliability is '98.41986455981942'. current run reliability is '85.71428571428571'. drift is 12.70558 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
| ShadowLinkingRandomOpsTest | test_node_operations | {"failures": true} | integration | https://buildkite.com/redpanda/redpanda/builds/75268#019a3294-febc-4156-b9e0-15b52f8b3ff4 | FLAKY         | 58/60    | upstream reliability is '99.54337899543378'. current run reliability is '90.9090909090909'. drift is 8.63429 and the allowed drift is set to 50. The test should PASS    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingRandomOpsTest&test_method=test_node_operations |
</details>
<!--/body-->

bharathv commented:
/ci-repeat 1
skip-redpanda-build
skip-units
dt-repeat=20
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv commented:
/ci-repeat 1
skip-redpanda-build
skip-units
dt-repeat=40
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv commented:
/ci-repeat 1
skip-redpanda-build
skip-units
dt-repeat=40
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv committed:
cl/link_mgr: relax replicator existence requirement in same term

bharathv committed:
cl/link_mgr: rework replicator start/stop dispatch.

Refactor how the replication_mgr handles start and stop notifications for
replicators.

Previously, all partition notifications on a shard were processed through a
single queue, which led to several issues:

 * Head-of-line blocking: starting a replicator may involve querying the
   source cluster for offsets (with retries), potentially blocking all
   subsequent notifications.
 * Duplicate work: multiple notifications for the same replicator (e.g.
   start -> stop -> start) were processed independently instead of being
   collapsed into a single effective target state.
 * Concurrency issues: under some conditions, multiple replicators could
   start in the same term, triggering assertion failures in RNOT.

This commit introduces a per-ntp reconciler (with a cap on concurrent
fibers). All in-flight requests for a given replicator are merged into a
single target state—either started or stopped—and the reconciler drives
the actual state toward that target, eliminating redundant work and
preventing blocking.

bharathv committed:
cl/replicator: introduce initiate_shutdown

.. to be used in the next commit

bharathv committed:
cl/replicator: introduce initiate_shutdown

.. to be used in the next commit

bharathv committed:
cl/replicator: make stop 2 step

First step initiates a shutdown (incase start() is stuck) and the actual
stop is enqueued.

This ensures any start() is not stuck forever causing head of line
blocking.

Future improvement is to have a reconciler/queue per ntp so one ntp does
not block another.

bharathv commented:
/ci-repeat 1
skip-redpanda-build
skip-units
dt-repeat=40
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv committed:
attempt one

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=40
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv committed:
debug try2

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=20
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=20
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=40
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv commented:
/ci-repeat 1
skip-units
dt-repeat=40
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

bharathv commented:
/ci-repeat 10
skip-redpanda-build
skip-units
dt-repeat=40
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75268-->
<!--bd3df9af-269e-4b97-bf6f-1365264b438c-->
<!--title-->
## Retry command for [Build#75268](https://buildkite.com/redpanda/redpanda/builds/75268)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/shadow_linking_rnot_test.py::ShadowLinkingRandomOpsTest.test_node_operations@{"failures":true}
```
<!--/body-->

bharathv committed:
Revert "attempt one"

This reverts commit 89a86772987956c2a279bdcaa086e2e42203c452.

bharathv committed:
Revert "cl/replicator: make stop 2 step"

This reverts commit 53a4965a6ba40938bd35938401b0e52a681236e0.

bharathv committed:
Revert "cl/replicator: introduce initiate_shutdown"

This reverts commit 0d6ee75fa7d461818602deab0071700183c5ed06.

bharathv committed:
Revert "cl/link_mgr: relax replicator existence requirement in same term"

This reverts commit 59e803af3826e4083a8a43eae983d1ae3169cce9.

bharathv committed:
defer updating snapshot offsets until log truncation
